storey/redis_driver.py

```diff
@@ -0,0 +1,482 @@
+                            if list_attribute_key not in redis_keys_involved:
+                                redis_keys_involved.append(list_attribute_key)
+                            lua_script = f'{lua_script}list_attribute_key="{list_attribute_key}";\n'
+
+                            if cached_time < expected_time:
+                                if not initialized_attributes.get(list_attribute_key, 0) == expected_time:
+                                    initialized_attributes[list_attribute_key] = expected_time
+                                    lua_script = f'{lua_script}local t=redis.call("GET","{array_time_attribute_key}");if (type(t)~="boolean" and (tonumber(t) < {expected_time})) \
+                                        then redis.call("DEL",list_attribute_key); end;\n'
+                                    default_value = self._convert_python_obj_to_redis_value(aggregation_value.default_value)
+                                    lua_script = f'{lua_script}len=redis.call("LLEN",list_attribute_key) for i=1,({bucket.total_number_of_buckets}-len) \
+                                        do redis.call("RPUSH",list_attribute_key,{default_value}) end;\n'
+                                if array_time_attribute_key not in times_updates:
+                                    times_updates[array_time_attribute_key] = expected_time_expr
+                                new_cached_times[name] = (array_time_attribute_key, expected_time)
+
+                            # Updating the specific cells
+                            if cached_time <= expected_time:
+                                lua_script = f'{lua_script}local old_value=redis.call("LINDEX",list_attribute_key,{index_to_update});\
+                                    old_value=tonum(old_value);\n'
+                                lua_script = f'{lua_script}old_value=tonum(old_value)\n'
+                                new_value_expression = aggregation_value.aggregate_lua_script('old_value', aggregation_value.value)
+                                lua_script = f'{lua_script}\
+                                    redis.call("LSET", list_attribute_key, {index_to_update}, {new_value_expression});\n'
+
+                        redis_keys_involved.append(array_time_attribute_key)
+                        lua_script = f'{lua_script}redis.call("SET","{array_time_attribute_key}",{expected_time}); \n'
+
+        return lua_script, condition_expression, pending_updates, redis_keys_involved
+
+    async def _save_key(self, container, table_path, key, aggr_item, partitioned_by_key, additional_data):
+        redis_key_prefix = self._make_key(container, table_path, key)
+        update_expression, mtime_condition, pending_updates, redis_keys_involved = self._build_feature_store_lua_update_script(
+            redis_key_prefix, aggr_item, partitioned_by_key, additional_data)
+        if not update_expression:
+            return
+        current_time = int(time.time_ns() / 1000)
```

Comment:
Microseconds on purpose?

Reply:
occurs twice in the code so I assume yes :-)
what is the concern? suggest keeping the mtime at nano resolution?

Reply:
Not really concerned about it, just wanted to make sure it's a deliberate choice :)